### DISCUSSION TITLE
I can't use Templated Queries

### DISCUSSION THREAD

**Discussion by firebabyIgnacio** [spaceuptech/space-cloud#1656] [Q&A] [answered]
**I'll follow the example(https://docs.space-cloud.io/storage/database/templated-queries/),But it didn't work out，**
![image](https://user-images.githubusercontent.com/73700924/219576842-a89de815-5049-4988-baed-9ed63bf82b03.png)
**I took a look at the logs**，
{"L":"INFO","T":"2023-02-17T15:18:56.512+0800","C":"server/middleware.go:28","M":"Request","requestId":"2Lr9Sql55gFGDCMKWNjiQzF7vfe","fields":{"body":"{\"operationName\":null,\"variables\":{\"userId\":\"1\",\"name\":\"Rock\"},\"query\":\"{\\n  changeName(userId: $userId, name: $name) @profile {\\n    status\\n    error\\n  }\\n}\\n\"}","method":"POST","queryVars":{},"url":"/v1/api/testapi/graphql"}}
{"L":"INFO","T":"2023-02-17T15:19:21.269+0800","C":"server/middleware.go:28","M":"Request","requestId":"2Lr9VvVGlxvR85q39wQD329wSRP","fields":{"body":"{\"query\":\"mutation {\\n  update_users (where: { id: $userId }, set: { name: $name }) @mongo {\\n    status\\n    error\\n  }\\n}\",\"variables\":null}","method":"POST","queryVars":{},"url":"/v1/api/testapi/graphql"}}
**I found that the second time on internal call has no parameters**
**The deployment environment is deployed following the tutorial(https://docs.space-cloud.io/install/docker/)(mysql)**

**Comment by YourTechBud** [comment, marked as answer]
Can you share the config of your templated query?

**Reply by firebabyIgnacio** [reply]
mutation {
  update_users (where: { id: $userId }, set: { name: $name }) @mongo{
    status
    error
  }
}
I followed the docs
https://docs.space-cloud.io/storage/database/templated-queries/

**Reply by YourTechBud** [reply]
You would have to add a template in the variable holder of the templated query as well.

```
{
  "userId": "{ .args.userId }",
  "name": "{ .args.name }"
}
```

**Reply by firebabyIgnacio** [reply]
i see, i think this should be added to the doc

**Reply by YourTechBud** [reply]
Will do. The next verstof SC is going to revamp the entire experience of using templated queries.

**Reply by firebabyIgnacio** [reply]
i found a problem when frequently building the same database，my database connection is full
step 1,no database
![image](https://user-images.githubusercontent.com/73700924/221773727-26c2bd24-2bf4-466a-893c-cd4ec6441d5b.png)
![image](https://user-images.githubusercontent.com/73700924/221773742-d0fd1b8e-d3c0-425e-af31-4355ccf18784.png)
step 2 add a database,i choose mysql,two connections have been added and the time is refreshed every ten seconds
![image](https://user-images.githubusercontent.com/73700924/221773761-64c12b85-08dd-4854-9c5f-94d695d0176a.png)
![image](https://user-images.githubusercontent.com/73700924/221773775-5b3d7b48-ace7-416f-a278-4be5aec32d76.png)
step 3 remove database,there is still a connection,when I create and remove databases multiple times, my database connection is full
![image](https://user-images.githubusercontent.com/73700924/221773802-f04ebb17-dceb-4c38-a559-82d3f282a93d.png)
![image](https://user-images.githubusercontent.com/73700924/221773807-552d93ac-105e-45ab-95e8-a763dd35ad58.png)
![image](https://user-images.githubusercontent.com/73700924/221773813-73969a1c-ec9b-457e-b130-b5842864e09a.png)

**Reply by YourTechBud** [reply]
That's interesting. Let me check this once

**Reply by firebabyIgnacio** [reply]
excuse me, but i also have a question,1. is every time a new database is created, will two connections be maintained all the time? if so, then i have a little more database, so isn't the number of connections to the database consumed quickly? 2.won't it automatically disconnect because there has been no operation for a long time?
in response to question 1:i found that in gateway\modules\crud\crud.go:89, the c here seems to be leaked and cannot be managed, so two connections are created.
2:gateway\modules\crud\sql\sql.go:68,there's a Ticker set to 10s, and it looks like this will keep the connection open forever

this is just a guess on my part (i'm not a Go developer)

**Reply by YourTechBud** [reply]
Yes. Currently each database would have its own connection pool. You'll need to use a database proxy if you want to multiplex connections to the database

**Reply by firebabyIgnacio** [reply]
got it, will this connection always be maintained? even if it is not used for a long time?
